gateway-ha/src/test/java/io/trino/gateway/ha/TestGatewayHaMultipleBackend.java

```diff
@@ -145,6 +145,11 @@ public void testQueryDeliveryToMultipleRoutingGroups()
                         .build();
         Response response1 = httpClient.newCall(request1).execute();
         assertThat(response1.body().string()).contains("http://localhost:" + routerPort);
+
+        List<Cookie> cookies = Cookie.parseAll(response1.request().url(), response1.headers());
+        Optional<Cookie> cookie = cookies.stream().filter(c -> c.name().equals("trinoClusterHost")).findAny();
+        assertThat(cookie).isNotEmpty();
```

Comment:
We should test for a specific value to make sure the cluster host is set correctly. You can control which backend the request is sent to using the `X-Trino-Routing-Group` header. When you set 
```
.addHeader("X-Trino-Routing-Group", "scheduled")
```
then the backend should be `http://localhost:backend2Port`, when `X-Trino-Routing-Group` is unset you'll get `http://localhost:backend1Port`. You should also use the test to define the behavior when the initial client request already contains a cookie with the key `trinoClusterHost` - I expect the value should be overwritten. It will be cleaner to make a new test method.